tools/lkl/lib/virtio_net_tap.c

```diff
@@ -122,5 +196,24 @@ struct lkl_netdev *lkl_netdev_tap_create(const char *ifname)
 		return NULL;
 	}
 
+	#ifdef CONFIG_EVENTFD
```

Comment:
I think the 500 fd limit with hijacking also comes up because of stuff like `fprintf(STDOUT, ...)`. It's confined to the hijacking code, though, so it's not too bad.

I understand your solution to be saying something like:

in `virtio_net_tap.{h,c}`:
```
struct lkl_tap_host_ops {
    .eventfd = eventfd, // refers to host eventfd if not hijacked, but LKL eventfd if hijacked
} lkl_tap_host_ops_vals;
```

and somewhere in `hijack/init.c`:

```
#import "virtio_net_tap.h"

__attribute__((constructor)) void fixup_lkl_tap_host_ops_vals(void)
{
    lkl_tap_host_ops_vals.eventfd = dlsym(RTLD_NEXT, "eventfd");
}
```

and "undo" the hijacking. Do I understand correctly?

It looks a little bit awkward to me to say `.eventfd = eventfd` since we'll never need that layer of indirection without hijacking. But it allows for this code to work as expected while `eventfd` is hijacked, and it provides more flexibility in the future if we want to swap other functions into this --- it might make unit testing significantly easier.

I had something in mind to provide `CONFIG_HIJACK_EVENTFD` and `CONFIG_CLEANUP_NETDEVS` as compile-time options and say that they're mutually exclusive, but I think your solution is better. I'll change this commit tomorrow.